### ISSUE TITLE
incorporate MM's propto stuff

### ISSUE THREAD

**Issue by bbolker** [glmmTMB/glmmTMB#931]
Maeve McGillycuddy has a new propto covstruct [over on this branch](https://github.com/glmmTMB/glmmTMB/compare/master...mmcgillycuddy:glmmTMB:prop2_covstruc).  Only tricky thing is to decide whether to add a nicer interface ...

**Comment by mmcgillycuddy** [comment]
Hi Ben, were you thinking of changing the interface for how the matrix is specified?

**Comment by bbolker** [comment]
I don't know. It would be nice to keep the matrix bundled with the primary data frame somehow. We do need to try to make sure that the matrix is available for downstream methods as necessary ...

**Comment by bbolker** [comment]
Also, I have a preference (but maybe not worth the structural challenges?) of passing data as *data*, rather than as mapped parameters (out of a concern that at some point we'll have to keep track of `map` settings that were done by user vs internally ...

**Comment by mmcgillycuddy** [comment]
Yes, that's a good point. I didn't want to make any structural changes to the inputs of TMB. I can make some changes and see if it works better?

**Comment by bbolker** [comment]
Mike Li and I have a [paper](https://github.com/wzmli/phyloglmm/blob/refactor/git_push/phyloglmm_ms_resubmit.pdf ) (submitted, have been working on revisions for a **long** time, hoping to resubmit ... where we simulated up to 500 species. We only went up to 500 species, which took 100 seconds or so to fit. Based on the scaling in this figure, unless something went wrong, another factor of 5 in size (~2500 species) should be about a 10-fold increase in fitting time, so about 1000 seconds ...

We are trying to solve approximately the same computational problem, but in a different way (we incorporate structure in the Z matrix rather than the covariance matrix); I haven't tried yet but am dying to look at the differences in computational efficiency for doing it different ways.

![Screenshot from 2023-10-25 11-25-45](https://github.com/glmmTMB/glmmTMB/assets/78918/484c0b5b-9705-43b0-adce-4a2be8661644)
